Don't convert NSURL to WKSecureCodingURLWrapper when encoding RVItem

achristensen07 opened the pull request:
#### 27067c78c1fbad65c4c13f71b7c879158483e9cd
<pre>
Don&apos;t convert NSURL to WKSecureCodingURLWrapper when encoding RVItem
<a href="https://bugs.webkit.org/show_bug.cgi?id=256680">https://bugs.webkit.org/show_bug.cgi?id=256680</a>
rdar://109237622

Reviewed by Wenson Hsieh.

RVItem&apos;s deserialization code deserializes an NSURL.  This is fine, except our encoder
has replaced it with a WKSecureCodingURLWrapper.  This is needed when encoding NSURLRequest
for standards compliance and it will be done here once rdar://109237983 is done, but in the
meantime just don&apos;t transform the NSURL when encoding.

* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(supportsRevealCore):
(-[WKSecureCodingArchivingDelegate archiver:willEncodeObject:]):
(-[WKSecureCodingArchivingDelegate init]):
(IPC::encodeSecureCodingInternal):
(IPC::shouldEnableStrictMode):

Canonical link: <a href="https://commits.webkit.org/264000@main">https://commits.webkit.org/264000@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ac4ebe84edbb48d1cc5dc212b4cc25bf46c06a33

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/6366 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/6565 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/6747 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/7935 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/6667 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/6790 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/6514 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/9563 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/6478 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/6472 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/5749 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/8012 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/3939 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/13608 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/5812 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/5817 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/8070 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/6302 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/5147 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/5708 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/9866 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/754 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/6078 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

whsieh commented on Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
```diff
@@ -68,11 +68,25 @@
+#if ENABLE(REVEAL)
+        result = PAL::isRevealCoreFrameworkAvailable();
+#else
+        UNUSED_PARAM(result);
```
I don't think these two lines are necessary.

achristensen07 committed:
Don't convert NSURL to WKSecureCodingURLWrapper when encoding RVItem
https://bugs.webkit.org/show_bug.cgi?id=256680
rdar://109237622

Reviewed by Wenson Hsieh.

RVItem's deserialization code deserializes an NSURL.  This is fine, except our encoder
has replaced it with a WKSecureCodingURLWrapper.  This is needed when encoding NSURLRequest
for standards compliance and it will be done here once rdar://109237983 is done, but in the
meantime just don't transform the NSURL when encoding.

* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(supportsRevealCore):
(-[WKSecureCodingArchivingDelegate archiver:willEncodeObject:]):
(-[WKSecureCodingArchivingDelegate init]):
(IPC::encodeSecureCodingInternal):
(IPC::shouldEnableStrictMode):

Canonical link: https://commits.webkit.org/264000@main
